custom_components/heatmiserneo/config_flow.py

```diff
@@ -37,31 +37,80 @@ class FlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
 
     def __init__(self):
         """Initialize Heatmiser Neo options flow."""
+        self._host = DEFAULT_HOST
+        self._port = DEFAULT_PORT
+        self._errors = None
+
+    async def async_step_zeroconf(self, discovery_info: DiscoveryInfoType):
+        """Handle zeroconf discovery."""
+        _LOGGER.debug("Zeroconfig discovered %s" % discovery_info)
+        self._host = discovery_info['hostname']
+
+        await self.async_set_unique_id(f"{self._host}:{self._port}")
+        self._abort_if_unique_id_configured()
+        return await self.async_step_zeroconf_confirm()
+
+    async def async_step_zeroconf_confirm(self, user_input=None):
+        _LOGGER.debug(f"context {self.context}")
+        """Handle a flow initiated by zeroconf."""
+        if user_input is not None:
+            self._errors = await self.try_connection()
+            if not self._errors:
+                return self._async_get_entry()
+            return await self.async_step_user()
 
+        return self.async_show_form(
+            step_id="zeroconf_confirm",
+            description_placeholders={
+                'name': self._host
+            },
+        )
 
+    async def try_connection(self):
+        _LOGGER.debug("Trying connection...")
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
Maybe `await asyncio.open_connection` is better here?

Reply:
Maybe but I say we leave it and then implement in a library with some kind of async hub.test()

Reply:
That's fine. At some point I can add it to https://git.stikonas.eu/andrius/neohub2 (alternatively merge requests are also welcome, at the moment I'm still adding other functionality)